### DISCUSSION TITLE
TASMOTA32 - BLE Thermometer to domoticz/mqtt

### DISCUSSION THREAD

**Discussion by adryb** [arendst/Tasmota#13801] [Q&A] [answered]
I successfully installed tasmota32-bluetooth on esp32 board, its correctly reading and showing data from lywsd03mmc bluetooth tehermometers (4 pcs) and i want to send this data to the Domoticz.

I cannot add sensor idx for this thermometers and information is not sent in format:
22:59:33.274 MQT: domoticz/in = {"idx":XX," ....
Its only sent in:
22:59:24.222 MQT: tele/tasmota_FA5D10/SENSOR = {"Time":"2021-11-25T22:59:24","ATC593cf3":{"mac":"a4c138593cf3","Temperature":20.8,"Humidity":35.9,"DewPoint":5.1,"Battery":94,"RSSI":-82}}

which is not seen by Domoticz.

How to export data from Bluetooth the same as from directly connected devices - this can be done by setting correct sensor idx'es.

**Comment by barbudor** [comment, marked as answer]
As of today there is no support in the BLE driver for Domiticz
You should be able to use Rules for capturing the output of the sensor and publish to domiticz/in
This example should provide you with some help : https://tasmota.github.io/docs/Rules/#use-of-variables-and-tele-in-domoticz
by replacing `tele-am2301-12` with `tele-ATC593cf3`

**Comment by adryb** [comment]
Thank you for reply. Rule added but nothing happens. 
Can you help me? I didnt make any advanced configs in Tasmota yet.


00:14:05.175 MQT: stat/tasmota_FA5D10/RESULT = {"Rule1":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":148,"Free":363,"Rules":"ON tele-ATC593cf3#temperature DO var1 %value% ENDON   ON tele-ATC593cf3#humidity DO publish domoticz/in {\"idx\":40,\"svalue\":\"%var1%;%value%;1\"} ENDON"}}
00:14:25.589 MQT: tele/tasmota_FA5D10/SENSOR = {"Time":"2021-11-26T00:14:25","ATC346304":{"mac":"a4c138346304","Temperature":21.5,"Humidity":52.3,"DewPoint":11.4,"Btn":1,"Battery":81,"RSSI":-80},"ATCb5ab71":{"mac":"a4c138b5ab71","Temperature":20.9,"Humidity":46.9,"DewPoint":9.1,"Btn":1,"Battery":100,"RSSI":-70}}
00:14:26.563 MQT: tele/tasmota_FA5D10/SENSOR = {"Time":"2021-11-26T00:14:26","ATCc9ef2f":{"mac":"a4c138c9ef2f","Temperature":21.1,"Humidity":46.2,"DewPoint":9.1,"Btn":1,"Battery":100,"RSSI":-80},"ATC593cf3":{"mac":"a4c138593cf3","Temperature":20.5,"Humidity":34.8,"DewPoint":4.4,"Btn":1,"Battery":93,"RSSI":-85}}
00:14:27.565 MQT: tele/tasmota_FA5D10/SENSOR = {"Time":"2021-11-26T00:14:27","ATC9fb51d":{"mac":"a4c1389fb51d","Temperature":21.6,"Humidity":44.2,"DewPoint":8.9,"Btn":1,"Battery":81,"RSSI":-67}}
00:14:35.542 MQT: tele/tasmota_FA5D10/STATE = {"Time":"2021-11-26T00:14:35","Uptime":"0T00:55:12","UptimeSec":3312,"Heap":82,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Berry":{"HeapUsed":3,"Objects":52},"Wifi":{"AP":1,"SSId":"Tech_WiFi","BSSId":"7A:45:58:28:83:65","Channel":6,"Mode":"11n","RSSI":100,"Signal":-33,"LinkCount":1,"Downtime":"0T00:00:06"}}
00:14:35.590 MQT: tele/tasmota_FA5D10/SENSOR = {"Time":"2021-11-26T00:14:35","ESP32":{"Temperature":50.0},"TempUnit":"C"}
00:14:35.609 MQT: tele/tasmota_FA5D10/BLE = {"Time":"2021-11-26T00:14:35","BLEDevices":{"total":9,"B9AC4CAFF7DB":{"i":0,"n":"BAC329956","r":-93},"A4C138346304":{"i":1,"r":-83},"A4C138B5AB71":{"i":2,"r":-70},"D45EEC3FB108":{"i":3,"r":-73},"A4C138C9EF2F":{"i":4,"r":-77},"A4C138593CF3":{"i":5,"r":-83},"E27F997309EA":{"i":6,"n":"G36","r":-91},"A4C1389FB51D":{"i":7,"r":-66},"5448E65755C8":{"i":8,"r":-71}}}
00:14:35.654 MQT: tele/tasmota_FA5D10/BLE = {"Time":"2021-11-26T00:14:35","BLE":{"scans":5,"adverts":32233,"devices":9,"resets":0}}
00:16:06.213 CMD: Rule   ON tele-ATC593cf3#Temperature DO var1 %value% ENDON   ON tele-ATC593cf3#Humidity DO publish domoticz/in {"idx":40,"svalue":"%var1%;%value%;1"} ENDON
00:16:06.234 RUL: Stored uncompressed, would compress from 148 to 110 (-26%)
00:16:06.249 MQT: stat/tasmota_FA5D10/RESULT = {"Rule1":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":148,"Free":363,"Rules":"ON tele-ATC593cf3#Temperature DO var1 %value% ENDON   ON tele-ATC593cf3#Humidity DO publish domoticz/in {\"idx\":40,\"svalue\":\"%var1%;%value%;1\"} ENDON"}}
00:18:00.148 CMD: rule1 1
00:18:00.166 MQT: stat/tasmota_FA5D10/RESULT = {"Rule1":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":148,"Free":363,"Rules":"ON tele-ATC593cf3#Temperature DO var1 %value% ENDON   ON tele-ATC593cf3#Humidity DO publish domoticz/in {\"idx\":40,\"svalue\":\"%var1%;%value%;1\"} ENDON"}}
00:19:25.552 MQT: tele/tasmota_FA5D10/SENSOR = {"Time":"2021-11-26T00:19:25","ATC346304":{"mac":"a4c138346304","Temperature":21.5,"Humidity":52.5,"DewPoint":11.4,"Btn":1,"Battery":81,"RSSI":-74},"ATCb5ab71":{"mac":"a4c138b5ab71","Temperature":20.9,"Humidity":46.9,"DewPoint":9.1,"Btn":1,"Battery":100,"RSSI":-71}}
00:19:26.561 MQT: tele/tasmota_FA5D10/SENSOR = {"Time":"2021-11-26T00:19:26","ATCc9ef2f":{"mac":"a4c138c9ef2f","Temperature":21.1,"Humidity":46.2,"DewPoint":9.0,"Btn":1,"Battery":100,"RSSI":-75},"ATC593cf3":{"mac":"a4c138593cf3","Temperature":20.5,"Humidity":34.7,"DewPoint":4.4,"Btn":1,"Battery":94,"RSSI":-83}}
00:19:27.559 MQT: tele/tasmota_FA5D10/SENSOR = {"Time":"2021-11-26T00:19:27","ATC9fb51d":{"mac":"a4c1389fb51d","Temperature":21.6,"Humidity":43.5,"DewPoint":8.7,"Btn":1,"Battery":81,"RSSI":-67}}
00:19:35.573 MQT: tele/tasmota_FA5D10/STATE = {"Time":"2021-11-26T00:19:35","Uptime":"0T01:00:12","UptimeSec":3612,"Heap":82,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Berry":{"HeapUsed":3,"Objects":52},"Wifi":{"AP":1,"SSId":"Tech_WiFi","BSSId":"7A:45:58:28:83:65","Channel":6,"Mode":"11n","RSSI":100,"Signal":-33,"LinkCount":1,"Downtime":"0T00:00:06"}}
00:19:35.627 MQT: tele/tasmota_FA5D10/SENSOR = {"Time":"2021-11-26T00:19:35","ESP32":{"Temperature":51.7},"TempUnit":"C"}
00:19:35.654 MQT: tele/tasmota_FA5D10/BLE = {"Time":"2021-11-26T00:19:35","BLEDevices":{"total":9,"B9AC4CAFF7DB":{"i":0,"n":"BAC329956","r":-92},"A4C138346304":{"i":1,"r":-74},"A4C138B5AB71":{"i":2,"r":-72},"D45EEC3FB108":{"i":3,"r":-73},"A4C138C9EF2F":{"i":4,"r":-74},"A4C138593CF3":{"i":5,"r":-83},"E27F997309EA":{"i":6,"n":"G36","r":-97},"A4C1389FB51D":{"i":7,"r":-67},"5448E65755C8":{"i":8,"r":-69}}}
00:19:35.703 MQT: tele/tasmota_FA5D10/BLE = {"Time":"2021-11-26T00:19:35","BLE":{"scans":5,"adverts":36082,"devices":9,"resets":0}}

**Reply by sfromis** [reply]
Not using BLE much, but to me it looks like the sensor data is published as it is sent from the sensor, not by TelePeriod. Try skipping the `tele-` prefix on the trigger to grab a temperature. This should make the trigger act on every incoming data message.

**Reply by adryb** [reply]
Yes, this helped! Thanks a lot!
